### PR TITLE
[SOAR-17247] MISP api updates

### PR DESCRIPTION
**Pull request by dlaverty-r7**
### Description

Describe the proposed changes:

Problem:
 Issue with connecting to the MISP server. 

Solution:
A number of changes have been made to the API and these have meant that connection to the server among other things are failing. As part of this PR I have update the version of Pymisp used, updated several API calls, removed a number of actions which appear to have been deprecated. Updated SDK and unit tests. 

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [x] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/misp/help.md [hunk 1]
```diff
@@ -1,35 +1,33 @@
 # Description
 
-[MISP](http://www.misp-project.org/) is an open source threat sharing platform.
-Gather, store and then find correlations of indicators of compromise. Quality of data is determined by the open source community.
-This plugin utilizes the [MISP API](https://circl.lu/doc/misp/automation/index.html) and leverages the [pymisp](https://github.com/CIRCL/PyMISP) library.
+MISP is an open source threat sharing platform
```
Comment (igorski-r7) [inline plugins/misp/help.md:3]: I suggest revering that description here

File: plugins/misp/komand_misp/actions/add_attribute/action.py [hunk 2]
```diff
@@ -24,15 +24,15 @@ def run(self, params={}):
...
+            attribute = item["Attribute"]
         except KeyError:
-            self.logger.error("Improperly formatted attribute")
-            raise PluginException(preset=PluginException.Preset.UNKNOWN)
+            self.logger.error(f"Unable to add attribute {item}")
+            raise PluginException(preset=PluginException.Preset.UNKNOWN, data=item)
+        except Exception as error:
+            self.logger.error(f"Error when adding attribute: {error}")
+            raise PluginException(preset=PluginException.Preset.UNKNOWN, data=error)
         return {"attribute": attribute}
```
Comment (igorski-r7) [inline plugins/misp/komand_misp/actions/add_attribute/action.py:38]: Suggest refactoring it with `Output.ATTRIBUTE`

File: plugins/misp/komand_misp/actions/add_sightings/action.py [hunk 3]
```diff
@@ -16,7 +16,7 @@ def __init__(self):
     def run(self, params={}):
         client = self.connection.client
         try:
-            item = client.set_sightings({"values": params.get("sightings")})
+            item = client.add_sighting({"values": params.get("sightings")})
```
Comment (igorski-r7) [inline plugins/misp/komand_misp/actions/add_sightings/action.py:19]:
```suggestion
            item = client.add_sighting({"values": params.get(Input.SIGHTINGS, [])})
```

File: plugins/misp/komand_misp/actions/create_an_event/action.py [hunk 4]
```diff
@@ -25,16 +25,18 @@ def run(self, params={}):
...
+                event={
+                    "distribution": dist[params.get("distribution")] or None,
+                    "threat_level_id": params.get("threat_level_id"),
+                    "analysis": params.get("analysis") or None,
+                    "info": params.get("info"),
+                    "date": None,
+                    "published": params.get("published"),
+                    "orgc_id": params.get("orgc_id") or None,
+                    "org_id": params.get("org_id") or None,
+                    "sharing_group_id": params.get("sharing_group_id") or None,
```
Comment (igorski-r7) [inline plugins/misp/komand_misp/actions/create_an_event/action.py:38]: I suggest updating `params.get("...")` with `params.get(Input.<parameter_name>)`

File: plugins/misp/plugin.spec.yaml [hunk 5]
```diff
@@ -3,17 +3,45 @@ extension: plugin
...
 title: MISP
-description: MISP is an open source threat sharing platform
-version: 5.0.3
+description: "[MISP](http://www.misp-project.org/) is an open source threat sharing platform. Gather, store and then find correlations of indicators of compromise. Quality of data is determined by the open source community. This plugin utilizes the [MISP API](https://circl.lu/doc/misp/automation/index.html) and leverages the [pymisp](https://github.com/CIRCL/PyMISP) library"
+version: 6.0.0
 connection_version: 5
 sdk:
   type: slim
-  version: 5
+  version: 6
```
Comment (cmcnally-r7) [inline plugins/misp/plugin.spec.yaml:11]: Cn we specify `6.1.0` here

File: plugins/misp/komand_misp/actions/add_attribute/action.py [hunk 6]
```diff
@@ -10,29 +10,29 @@ class AddAttribute(insightconnect_plugin_runtime.Action):
...
-        event = params.get("event")
-        type_value = params.get("type_value")
-        category = params.get("category")
-        value = params.get("value")
-        comment = params.get("comment")
+        event = params.get(Input.EVENT)
+        type_value = params.get(Input.TYPE_VALUE)
+        category = params.get(Input.CATEGORY)
+        value = params.get(Input.VALUE)
+        comment = params.get(Input.COMMENT)
```
Comment (cmcnally-r7) [inline plugins/misp/komand_misp/actions/add_attribute/action.py:23]: since this isnt required can we add a default value here

**Review by cmcnally-r7 — CHANGES_REQUESTED** [review]

**Review by cmcnally-r7 — APPROVED** [review]

File: plugins/misp/plugin.spec.yaml [hunk 7]
```diff
@@ -3,17 +3,45 @@ extension: plugin
 products: [insightconnect]
 name: misp
 title: MISP
-description: MISP is an open source threat sharing platform
-version: 5.0.3
+description: "[MISP](http://www.misp-project.org/) is an open source threat sharing platform. Gather, store and then find correlations of indicators of compromise. Quality of data is determined by the open source community. This plugin utilizes the [MISP API](https://circl.lu/doc/misp/automation/index.html) and leverages the [pymisp](https://github.com/CIRCL/PyMISP) library"
+version: 6.0.0
 connection_version: 5
```
Comment (ekelly-r7) [inline plugins/misp/plugin.spec.yaml:8]: does connection_version need bumped on a major release?

Comment (rbowden-r7) [reply plugins/misp/plugin.spec.yaml:8]: no the connection version will only need bumped if there is a change to the connection object 

eg went from taking a name and pass to now using an api key

Comment (ekelly-r7) [reply plugins/misp/plugin.spec.yaml:8]: ahh ok, good to know!

**Review by ekelly-r7 — APPROVED** [review]

**Pull request merged**